x-pack/legacy/plugins/siem/public/containers/detection_engine/signals/api.test.ts

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { KibanaServices } from '../../../lib/kibana';
+import {
+  signalsMock,
+  mockSignalsQuery,
+  mockStatusSignalQuery,
+  mockSignalIndex,
+  mockUserPrivilege,
+} from './mock';
+import {
+  fetchQuerySignals,
+  updateSignalStatus,
+  getSignalIndex,
+  getUserPrivilege,
+  createSignalIndex,
+} from './api';
+
+const abortCtrl = new AbortController();
+const mockKibanaServices = KibanaServices.get as jest.Mock;
+jest.mock('../../../lib/kibana');
+
+const fetchMock = jest.fn();
+mockKibanaServices.mockReturnValue({ http: { fetch: fetchMock } });
+
+describe('Detections Signals API', () => {
+  describe('fetchQuerySignals', () => {
+    beforeEach(() => {
+      fetchMock.mockClear();
+      fetchMock.mockResolvedValue(signalsMock);
+    });
+
+    test('check parameter url, body', async () => {
+      await fetchQuerySignals({ query: mockSignalsQuery, signal: abortCtrl.signal });
+      expect(fetchMock).toHaveBeenCalledWith('/api/detection_engine/signals/search', {
+        body:
+          '{"aggs":{"signalsByGrouping":{"terms":{"field":"signal.rule.risk_score","missing":"All others","order":{"_count":"desc"},"size":10},"aggs":{"signals":{"date_histogram":{"field":"@timestamp","fixed_interval":"81000000ms","min_doc_count":0,"extended_bounds":{"min":1579644343954,"max":1582236343955}}}}}},"query":{"bool":{"filter":[{"bool":{"must":[],"filter":[{"match_all":{}}],"should":[],"must_not":[]}},{"range":{"@timestamp":{"gte":1579644343954,"lte":1582236343955}}}]}}}',
```

Comment:
These would be better as:

```ts
body: JSON.stringify({ aggs: ... }
```

Rather than string dumps for future maintainers.

Reply:
Very true; these mainly came over from my cherry-pick of X's branch, and I simplified them since they're no longer dealing with the try/catch logic. I think we'd find that this test is pretty much the same as the one below it, and that this argument is in fact `{ body: JSON.stringify(mockSignalsQuery) }` 😉